### PR TITLE
CMake: Support fine grained feature configuration constraints in tests

### PR DESCRIPTION
**Pull request by tamiko**
With this change it is now possible to restrict tests to fine grained
feature configurations (if we happen to export them to
deal.IIConfig.cmake).

Example:
 test.with_arpack=true.with_arpack_with_parpack=true.output

 Will only be configured if arpack is detected with (working) parpack
 support.

### CODE REVIEW

File: cmake/modules/FindPETSC.cmake [hunk 1]
```diff
@@ -52,39 +53,23 @@ DEAL_II_FIND_PATH(PETSC_INCLUDE_DIR_ARCH petscconf.h
...
-  IF("${PETSC_COMPLEX_STRING}" STREQUAL "")
-    SET(PETSC_WITH_COMPLEX FALSE)
-  ELSE()
-    SET(PETSC_WITH_COMPLEX TRUE)
-  ENDIF()
+IF(EXISTS ${PETSC_PETSCCONF_H})
+  _petsc_feature_check(64BIT_INDICES "#define.*PETSC_USE_64BIT_INDICES 1")
+  _petsc_feature_check(COMPLEX "#define.*PETSC_USE_COMPLEX 1")
+  _petsc_feature_check(MPIUNI "#define.*PETSC_HAVE_MPIUNI 1")
+  _petsc_feature_check(MUMPS "#define.*PETSC_HAVE_MUMPS 1")
```
Comment (davydden) [inline cmake/modules/FindPETSC.cmake]: maybe also pick-up superlu-dist ?

File: doc/news/changes.h [hunk 2]
```diff
@@ -150,6 +150,14 @@ inconvenience this causes.
 <h3>General</h3>
 
 <ol>
+ <li> Improved: The testsuite now supports fine grained feature constraints
```
Comment (davydden) [inline doc/news/changes.h]: you also pick-up extra config of PETSc

**Comment by davydden** [comment]
Looks good to me.

File: cmake/modules/FindPETSC.cmake (hunk 1 above)
Comment (tamiko) [inline cmake/modules/FindPETSC.cmake]: Do we currently have any deal.II tests that use superlu?

File: doc/news/changes.h (hunk 2 above)
Comment (tamiko) [inline doc/news/changes.h]: Jup. But I won't comment in changes.h on such a tiny change :-]

File: cmake/modules/FindPETSC.cmake (hunk 1 above)
Comment (davydden) [inline cmake/modules/FindPETSC.cmake]: probably not

Comment (tamiko) [inline cmake/modules/FindPETSC.cmake]: Then, there is no need for this check at the moment :-D

**Comment by tamiko** [comment]
@davydden *ping*

**Pull request merged**